06x-new-kafka-roller.md

```diff
@@ -0,0 +1,231 @@
+# Kafka Roller 2.0
+
+## Current situation
+
+The Kafka Roller is a Cluster Operator component that's responsible for coordinating the rolling restart or reconfiguration of Kafka pods when:
+- non-dynamic reconfigurations needs to be applied
+- update in Kafka CRD is detected
+- a certificate is renewed
+- pods have been manually annotated by the user for controlled restarts
+- pod is stuck and is out of date
+- Kafka broker is unresponsive to Kafka Admin connections
+
+These are not the exhaustive list of possible triggers for rolling Kafka pods, but the main ones to highlight.
+
+A pod is considered stuck if it is in one of following states:
+- `CrashLoopBackOff`
+- `ImagePullBackOff`
+- `ContainerCreating`
+- `Pending` and `Unschedulable`
```

Comment:
Actually we don't check how long it's been in Pending state. If we find that the pod is in Pending state and has an old revision, we restart it, otherwise the pod is skipped anyway.

Reply:
Yes I agree with this. I think with the current roller, the assumption is that we restarted the pod and waited for it to become ready for 5 mins and timed out. When timed out, the pod is retried and then if the pod is not ready and  in `ContainerCreating` state, we consider it stuck. Also the stuck pod is only retried if the revision has an old revision (e.g pod image is updated, we may need to restart it anyway). 

With the new roller, we do have a wait for pods in `ContainerCreating` state. I did not specifically explain this but make this clearer in the proposed changes section.